### DISCUSSION TITLE
exclusive packet capture mode

### DISCUSSION THREAD

**Discussion by JustinAzoff** [zeek/zeek#1234] [Ideas]
I think zeek should have a mode like -w, but where it ONLY runs the packet source plugin, and maybe events so something like misc/trim-trace-file.zeek works.

The benefit of this is that a user could use zeek for capturing a pcap from any installed packet source, without any of the packet processing and script overhead zeek -w has.

I think in theory a patch could look something like this

```
 void NetSessions::NextPacket(double t, const Packet* pkt)
        {
+       // Check whether packet should be recorded based on session analysis
+       if ( zeek::run_state::capture_only )
+               {
+               DumpPacket(pkt);
+               return;
+               }
```

but it might be better to have a dedicated mode for this, if there's any overhead in adding a branch at the top of every call to NextPacket..

**Comment by ckreibich** [comment]
I like this a lot — Zeek as an abstraction layer over supported packet capture mechanisms. Depending on the features offered by the package source you could still be selective about capturing specific streams/rings/pipes/etc or more broadly, it's not normally a given that in a particular installation tcpdump supports the same capture mechanisms, and alternative packet sources might offer rough tcpdump equivalents but you'll need to figure them out on a case-by-case basis.
